docs/RtspStreamingGuide.md

```diff
@@ -0,0 +1,371 @@
+
+## Overview
+EdgeX device service for communicating with USB cameras attached to Linux OS platforms.
+This service provides the following capabilities:
+- V4L2 API to get camera metadata.
+- Camera status
+- Video stream reference
+- FFmpeg framework to capture video frames and stream them to an RTSP server.
+- An [RTSP server](https://github.com/aler9/rtsp-simple-server) is embedded in the dockerized device service. 
+
+## Tested Devices
+The following devices have been tested with EdgeX:
+<!-- sorted alphabetically -->
+- AUKEY PC-LM1E Webcam
+- HP w200 Webcam
+- Jinpei JW-01B USB FHD Web Computer Camera
+- Logitech Brio 4K
+- Logitech C270 HD Webcam
+- Logitech StreamCam
+
+## Dependencies
+The software has dependencies, including Git, Docker, Docker Compose, and assorted command line tools. Follow the instructions below to install any dependency that is not already installed.  
+
+### Install Git
+Install Git from the official repository as documented on the [Git SCM](https://git-scm.com/download/linux) site.
+
+1. Update installation repositories:
+   ```bash
+   sudo apt update
+   ```
+
+2. Add the Git repository:
+   ```bash
+   sudo add-apt-repository ppa:git-core/ppa -y
+   ```
+
+3. Install Git:
+   ```bash
+   sudo apt install git
+   ```
+
+### Install Docker
+Install Docker from the official repository as documented on the [Docker](https://docs.docker.com/engine/install/ubuntu/) site.
+
+### Verify Docker
+To enable running Docker commands without the preface of sudo, add the user to the Docker group. Then run Docker with the `hello-world` test.
+
+1. Create Docker group:
+   ```bash
+   sudo groupadd docker
+   ```
+   >NOTE: If the group already exists, `groupadd` outputs a message: **groupadd: group `docker` already exists**. This is OK.
+
+2. Add User to group:
+   ```bash
+   sudo usermod -aG docker $USER
+   ```
+
+3. Refresh the group:
+   ```bash
+   newgrp docker 
+   ```
+
+4. To verify the Docker installation, run `hello-world`:
+
+   ```bash
+   docker run hello-world
+   ```
+   A **Hello from Docker!** greeting indicates successful installation.
+
+   ```bash
+   Unable to find image 'hello-world:latest' locally
+   latest: Pulling from library/hello-world
+   2db29710123e: Pull complete 
+   Digest: sha256:10d7d58d5ebd2a652f4d93fdd86da8f265f5318c6a73cc5b6a9798ff6d2b2e67
+   Status: Downloaded newer image for hello-world:latest
+
+   Hello from Docker!
+   This message shows that your installation appears to be working correctly.
+   ...
+   ```
+
+### Install Docker Compose
+Install Docker from the official repository as documented on the [Docker Compose](https://docs.docker.com/compose/install/#install-compose) site. See the Linux tab. 
+
+1. Download current stable Docker Compose:
+   ```bash
+   sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
```

Comment:
Hey folks, I just learned that docker invocation changed from "docker-compose" to "docker compose" without the hyphen. Just an FYI.

Reply:
Also, they seem to be up to 2.29 now, so 1.29.2 seems a bit old...

Reply:
Doing just an install from the repository results in and issue between all of the makefiles using docker compose, as the command for that is `docker compose`. I think @ajcasagrande has a fix, but I just reverted to 1.29.2 to fix this.

Reply:
for ubuntu 20.04 I still needed docker-compose installed